src/lib/node.ts

```diff
@@ -209,3 +209,75 @@ export function pipeToNodeStream<T>(
 		handleTsStreamError // abort handler
 	);
 }
+
+/**
+ * Create a ts-stream ReadableStream from a Node.JS ReadableStream.
+ *
+ * Reads all values from `nodeReadable` and writes them to the returned stream.
+ * When the `nodeReadable` ends, the returned stream is also ended.
+ * Should the `nodeReadable` emit a `close` event, the returned stream is ended without error.
+ * If the `nodeReadable` emits a `error` event, the returned stream is ended with the given error.
+ *
+ * Usage example:
+ * ```ts
+ * const source = fromNodeReadable(fs.createReadStream('source.txt'));
+ * source
+ *  .map((chunk) => String(chunk))
+ *  .map((letters) => letters.toUpperCase())
+ *  .forEach(console.log);
+ * ```
+ *
+ * @param nodeReadable Soource NodeJS stream
+ * @returns Readable stream of type `string | Buffer`
+ */
+export function fromNodeReadable(
+	nodeReadable: NodeJS.ReadableStream
+): ReadableStream<string | Buffer> {
+	const stream = new Stream<string | Buffer>();
+
+	const endStream = async (err?: Error) => {
+		if (!stream.isEndingOrEnded()) return stream.end(err);
+	};
+
+	stream.aborted().catch(endStream);
+	nodeReadable.once("close", endStream);
+	nodeReadable.once("error", endStream);
+
+	(async () => {
+		try {
+			for await (const chunk of nodeReadable) {
+				await stream.write(chunk);
+			}
+			await endStream();
+		} catch (err) {
+			await endStream(err as Error);
+		}
+	})();
```

Comment:
This IIFE can cause an unhandled promise rejection exception (if `stream.end()` fails).
Not sure what to do about it though. Perhaps it makes sense to try and 'back-fire' it as an error event on the readable.

For example, if `stream.write()` fails, and subsequently `stream.end(err)` also returns an error.

Reply:
Hmm, that's a good point. I think it's appropriate to emit an error event on the `nodeReadable` since we'll only attempt to act upon an error even on that node stream once here in this function.

Reply:
Updated to emit an error should `steam.end` reject in the `endStream` helper above.

Reply:
Yes, good solution to handle it in the helper!